lib/resources/elasticsearch.rb

```diff
@@ -0,0 +1,168 @@
+      configuration settings. The data is extract from the live elastic search
+      cluster using the curl command `curl localhost:9200/_nodes/`. The resource
+      makes uses of FilterTable and its functionality can be used "
+
+    example "
+      describe elasticsearch(es_ip:'0.0.0.0', es_port:'9200', es_user:'elastic', es_pass:'changeme', https:true, self_signed_cert:true).nodes do
+        it { should exist }
+      end
+
+      describe elasticsearch.nodes do
+        its('name') { should include 'my_node' }
+      end
+
+      describe elasticsearch.nodes do
+        its('name') { should_not include 'node_name' }
+      end
+
+      elasticsearch.nodes.os.each do |nodeos|
+        describe nodeos do
+          its('name') { should_not cmp 'MacOS' }
+        end
+      end
+
+      elasticsearch.nodes.version.each do |node_version|
+        describe node_version do
+          it { should be > '1.2.0' }
+        end
+      end
+
+      elasticsearch.nodes.where {os.name == 'Linux' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') { should cmp '/etc/elasticsearch' }
+        end
+      end
+
+      elasticsearch.nodes.where {name == 'package-centos-72' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') { should cmp '/etc/elasticsearch' }
+        end
+      end
+        "
+
+    def initialize(opts = {}) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+      return skip_resource 'Package `curl` not avaiable on the host' unless inspec.command('curl').exist?
+
+      es_ip            = opts.fetch(:es_ip, '0.0.0.0')
+      es_port          = opts.fetch(:es_port, '9200')
+      es_user          = opts.fetch(:es_user, nil)
+      es_pass          = opts.fetch(:es_pass, nil)
+      https            = opts.fetch(:https, true)
+      self_signed_cert = opts.fetch(:self_signed_cert, false)
+
+      url = "-H 'Content-Type: application/json' "
+      url += https ? 'https' : 'http'
+      url += "://#{es_ip}:#{es_port}/_nodes/"
+      url += " -u #{es_user}:#{es_pass}" unless es_user.nil? || es_pass.nil?
+      url += ' -k' if self_signed_cert.eql?(true)
+
+      cmd = inspec.command("curl #{url}")
+      return skip_resource 'Connection refused please check ip and port provided' if cmd.stderr =~ /Failed to connect/
+      return skip_resource "Connection refused Peer's Certificate issuer is not recognized" if cmd.stderr =~ /Peer's Certificate issuer is not recognized/
+      return skip_resource "Error fetching Elastcsearch data from curl #{url}: #{cmd.stderr}" unless cmd.exit_status.zero?
+
+      begin
+        @content = JSON.parse(cmd.stdout)
+      rescue JSON::ParserError => e
+        return skip_resource "Couldn't parse ES data: #{e.message}"
+      end
+
+      return skip_resource "Security Exception: #{@content['error']}" unless @content['error'].nil?
```

Comment:
Can you explain this? What's a "security exception"?

Reply:
I have modified it to show status of the error from elasticsearch